src/dataLayer/databaseAttachmentManager.js

```diff
@@ -0,0 +1,106 @@
+const Readable = require('stream').Readable;
+
+const { assertTypeEquals } = require('../utils/assertType');
+const { MongoDatabaseManager } = require('../utils/mongoDatabaseManager');
+const { ConfigManager } = require('../utils/configManager');
+const Attachment = require('../fhir/classes/4_0_0/complex_types/attachment');
+
+/**
+ * @classdesc This class handles attachments with Mongodb GridFS i.e. converts attachment.data
+ * @classdesc to attachment._file_id and restores the attachment._file_id to attachment.data
+*/
+class DatabaseAttachmentManager {
+    /**
+     * constructor
+     * @param {MongoDatabaseManager} mongoDatabaseManager
+     * @param {ConfigManager} configManager
+    */
+    constructor({mongoDatabaseManager, configManager}) {
+        /**
+         * @type {MongoDatabaseManager}
+        */
+        this.mongoDatabaseManager = mongoDatabaseManager;
+        assertTypeEquals(mongoDatabaseManager, MongoDatabaseManager);
+
+        /**
+         * @type {ConfigManager}
+        */
+        this.configManager = configManager;
+        assertTypeEquals(configManager, ConfigManager);
+    }
+
+    /**
+     * Checks if GridFs is applicable on the resource and if applicable applies GridFs
+     * @param {Resource[]|Resource} resources
+    */
+    async transformAttachments(resources) {
+        const enabledGridFsResources = this.configManager.enabledGridFsResources;
+        if (Array.isArray(resources)) {
+            for (let resourceIndex = 0; resourceIndex < resources.length; resourceIndex++) {
+                const resource = resources[parseInt(resourceIndex)];
+                if (enabledGridFsResources.includes(resource.resourceType)) {
+                    resources[parseInt(resourceIndex)] =
+                        await this.changeAttachmentWithGridFS(
+                            resource,
+                            resource.id,
```

Comment:
Let's pass the metadata object directly from here consisting of resource._uuid and resource._sourceId instead of resource.id.